packages/flutter/test/material/chip_test.dart

```diff
@@ -130,6 +132,66 @@ Future<void> _testConstrainedLabel(
   expect(chipSize.height, chipParentHeight);
 }
 
+Future<void> _pumpCheckmarkChip<T>(
+    WidgetTester tester, {
+      Color themeColor,
+      Color color,
+      Brightness brightness = Brightness.light
+    }) async {
+  Widget chip;
+  if (T == InputChip) {
+    chip = InputChip(
+      label: const Text('InputChip'),
+      selected: true,
+      showCheckmark: true,
+      checkmarkColor: color,
+    );
+  } else if (T == FilterChip) {
+    chip = FilterChip(
+      label: const Text('FilterChip'),
+      selected: true,
+      showCheckmark: true,
+      checkmarkColor: color,
+      onSelected: (bool _) { },
+    );
+  } else {
+    throw UnsupportedError('Chip type must be InputChip of FilterChip');
```

Comment:
Rather than checking the type of the generic `T`, consider having this function take a `Widget chip` parameter, then make separate functions for creating the chip types.

This would keep you from having to 1) check the type and 2) throw an error when there is an "unsupported" chip type. 

Calling this function would then look something like:

```dart
_pumpCheckmarkChip(tester, themeColor: Color(...), chip: _filterChip(checkMarkColor: Color(...)));
```

Reply:
Good idea! done